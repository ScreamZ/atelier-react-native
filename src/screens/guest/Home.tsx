import React, { useEffect } from "react";
import { Text, SafeAreaView, Button } from "react-native";
import { NavigationScreenProp, NavigationState } from "react-navigation";
import { Screens } from "../../modules/navigation/types";

export const HomeScreen: React.FC<{ navigation: NavigationScreenProp<NavigationState> }> = props => {
  return (
    <SafeAreaView>
      <Text>Hello homescreen</Text>
      <Button onPress={() => props.navigation.navigate(Screens.GuestLogin)} title="Go to login" />
    </SafeAreaView>
  );
};
