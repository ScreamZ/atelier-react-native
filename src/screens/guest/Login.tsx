import React from "react";
import { SafeAreaView, Text } from "react-native";
import { NavigationScreenComponent } from "react-navigation";

const LoginScreen: NavigationScreenComponent = () => {
  return (
    <SafeAreaView>
      <Text>Hello login</Text>
    </SafeAreaView>
  );
};

export default LoginScreen;
