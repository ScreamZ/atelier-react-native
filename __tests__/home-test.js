import React from 'react'
import { HomeScreen } from '../src/screens/guest/Home'

import renderer from 'react-test-renderer'

test('Is home rendered correctly', () => {
  const tree = renderer.create(<HomeScreen />).toJSON()
  expect(tree).toMatchSnapshot()
});
