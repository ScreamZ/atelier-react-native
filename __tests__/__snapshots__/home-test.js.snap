// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Is home rendered correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <Text>
    Hello homescreen
  </Text>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Go to login
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
